source/common/stats/thread_local_store.cc

```diff
@@ -217,12 +217,41 @@ std::atomic<uint64_t> ThreadLocalStoreImpl::ScopeImpl::next_scope_id_;
 
 ThreadLocalStoreImpl::ScopeImpl::~ScopeImpl() { parent_.releaseScopeCrossThread(this); }
 
+bool ThreadLocalStoreImpl::checkAndRememberRejection(const std::string& name,
+                                                     ConstCharStarHashSet* tls_rejected_stats) {
+  if (stats_matcher_->acceptsAll()) {
+    return false;
+  }
+
+  const char* rejected_name = rejected_stats_.find(name);
+  if (rejected_name == nullptr) {
+    if (rejects(name)) {
+      rejected_name = rejected_stats_.insert(name);
+    }
+  }
+  if (rejected_name != nullptr) {
+    if (tls_rejected_stats != nullptr) {
+      tls_rejected_stats->insert(rejected_name);
+    }
+    return true;
+  }
+  return false;
+}
```

Comment:
This makes sense but seems easy to mangle. Is there a way to extract the component of this which checks and/or writes into rejected_stats_ from the component which writes into tls_rejected_stats?

Reply:
I think if I break this up it will be harder to use in its multiple call-sites. I attempted to do mangle-protection with tests. WDYT?

Reply:
SGTM, that's a great reason not to break up this function.